core/app/[locale]/(default)/(auth)/login/_actions/login.ts

```diff
@@ -22,12 +32,13 @@ export const login = async (_previousState: unknown, formData: FormData) => {
       redirect: false,
     });
 
-    redirect({ href: '/account', locale });
-  } catch (error: unknown) {
-    rethrow(error);
+    return redirect({ href: '/account', locale });
+  } catch (error) {
+    // We need to throw this error to trigger the redirect as Next.js uses error boundaries to redirect.
+    if (isRedirectError(error)) {
+      throw error;
+    }
 
-    return {
-      status: 'error',
-    };
+    return submission.reply({ formErrors: [t('Form.error')] });
   }
```

Comment:
I think we can avoid having to use Next.js internals here by moving the redirect to be _after_ the try/catch

Reply:
https://nextjs.org/docs/app/building-your-application/routing/redirecting#redirect-function

Reply:
![CleanShot 2024-12-05 at 14 37 30@2x](https://github.com/user-attachments/assets/6d16785a-c003-4603-bf16-d6b1988019ed)

Fixed!